controlplane/pkg/model/model.go

```diff
@@ -146,10 +178,14 @@ func (i *impl) DeleteClientConnection(connectionId string) {
 	clientConnection := i.clientConnections[connectionId]
 	if clientConnection == nil {
 		i.Unlock()
```

Comment:
Can we just `defer i.Unlock()` while we're changing here?

Reply:
Nope, defer is not working, since firstly it lock for writing and for listeners lock for reading.